flashtext/keyword.py

```diff
@@ -482,7 +457,7 @@ def extract_keywords(self, sentence, span_info=False):
         while idx < sentence_len:
             char = sentence[idx]
             # when we reach a character that might denote word end
-            if char not in self.non_word_boundaries:
+            if KeywordProcessor.NON_WORD_CHAR_REGEX.match(char):
```

Comment:
why the ugly direct reference to the class? just use `self`

Reply:
also, the original code checks if `char` IS a word char, your code checks if it's NOT a word char